cartography/intel/okta/oktaintel.py

```diff
@@ -0,0 +1,1225 @@
+    """
+    Add trusted origins to the graph
+    :param neo4j_session: session with the Neo4j server
+    :param okta_org_id: okta organization id
+    :param trusted_list: list of trusted origins
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    ingest = """
+    MATCH (org:OktaOrganization{id: {ORG_ID}})
+    WITH org
+    UNWIND {TRUSTED_LIST} as data
+    MERGE (new:OktaTrustedOrigin{id: data.id})
+    ON CREATE SET new.firstseen = timestamp()
+    SET new.name = data.name,
+    new.origin = data.origin,
+    new.scopes = data.scoped,
+    new.status = data.status,
+    new.created = data.created,
+    new.created_by = data.created_by,
+    new.okta_last_updated = data.okta_last_updated,
+    new.okta_last_updated_by = data.okta_last_updated_by,
+    new.lastupdated = {okta_update_tag}
+    WITH org, new
+    MERGE (org)-[r:RESOURCE]->(new)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest,
+        ORG_ID=okta_org_id,
+        TRUSTED_LIST=trusted_list,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def _sync_trusted_origins(neo4j_session, okta_org_id, okta_update_tag):
+    """
+    Sync trusted origins
+    :param neo4j_session: session with the Neo4j server
+    :param okta_org_id: okta organization id
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    logger.debug("Syncing Okta Trusted Origins")
+
+    api_client = _create_api_client(okta_org_id, "/api/v1/trustedOrigins")
+
+    trusted_list = _get_trusted_origins(api_client)
+    _ingest_trusted_origins(neo4j_session, okta_org_id, trusted_list, okta_update_tag)
+
+
+def sync(neo4j_session, config):
+    """
+    Starts the Okta ingestion process by initializing Okta API session and pulling necessary information
+    :param neo4j_session: The Neo4j session
+    :param config: A `cartography.config` object
+    :return: Nothing
+    """
+
+    okta_organization = config.okta_organization
+
+    logger.debug(f"Starting Okta sync on {okta_organization}")
+
+    common_job_parameters = {
+        "UPDATE_TAG": config.update_tag,
+        "OKTA_ORG_ID": okta_organization,
+    }
+
+    _create_okta_organization(neo4j_session, okta_organization, config.update_tag)
+    _sync_okta_users(neo4j_session, okta_organization, config.update_tag)
+    _sync_okta_groups(neo4j_session, okta_organization, config.update_tag)
+    _sync_okta_applications(session, okta_organization, last_update)
+    _sync_users_factors(session, okta_organization, last_update)
+    _sync_trusted_origins(session, okta_organization, last_update)
+
+    # need creds with permission
+    # soft fail as some won't be able to get such high priv token
+    try:
+        _sync_roles(session, okta_organization, last_update)
+    except Exception as exception:
```

Comment:
Recommend against catching the base exception - keep this scoped to only exceptions that we know will happen.  What is the name of exception if the token has insufficient privs?

Reply:
fixed